### DISCUSSION TITLE
Problems scraping youtube playlists!

### DISCUSSION THREAD

**Discussion by Quills245** [pystardust/ytfzf#479] [Q&A] [answered]
I wanted to know if its normal for it to take this long for scrapping a youtube playlist, im using the command `ytfzf --type=playlist <playlist name>` but it keeps saying `Scraping Youtube playlist (with https://vid.puffyan.us) (playlist: <playlist link>, pg: 1)` for a really long time, when bigger the playlist more time it takes, in a playlist with just 6 videos it took a bit more than 6 minutes to scrap and it even gave me an error in the end, but im not sure if the error always appear or was just in this case, later i was trying a bigger playlist and it was tooking forever! until i get upset and cancel it.

**Comment by Euro20179** [comment]
You might want to try changing invidious instances.

you can add `invidious_instance="link_to_new_instace"` in `~/.config/ytfzf/conf.sh` to do this.

Some instances do not expose their api, those won't work

I changed mine to "https://invidious.namazso.eu" recently and it works well.
But if you want something else [here is the link of official instances](https://api.invidious.io)
If you want you can even host your own instance and set invidious_instance to `http://localhost:<port-that-invidious-is-on>`

EDIT:
`--type=playlist` only makes it so it searches for playlists

You may want to try `ytfzf -c invidious-playlist <link to playlist>`

**Reply by Quills245** [reply]
oh! ok i will try this, thanks!

**Reply by Quills245** [reply]
> EDIT: `--type=playlist` only makes it so it searches for playlists

yeah, this problem happen after i choose one playlist from the search

**Reply by Quills245** [reply]
> You might want to try changing invidious instances.
> 
> you can add `invidious_instance="link_to_new_instace"` in `~/.config/ytfzf/conf.sh` to do this.
>
> I changed mine to "https://invidious.namazso.eu" recently and it works well.
> 
> You may want to try `ytfzf -c invidious-playlist <link to playlist>`

i tryed doing this things but, just same thing...

**Comment by Euro20179** [comment, marked as answer]
> but it keeps saying Scraping Youtube playlist (with https://vid.puffyan.us) (playlist: <playlist link>, pg: 1) for a really long time

Ahhh yes, so basically the only way to tell if it's at the last page of a playlist is to check if it gives an empty list, if it never does, it scrapes forever.

what is the output of `ytfzf --version`, if it's 2.0 or lower it may be worth upgrading to 2.1-rc4 (which is the stable version of 2.1, I just forgot to change the version upon release lol) or higher.

**Reply by Quills245** [reply]
It says `ytfzf: 2.1.rc-4` so i think its updated...

**Reply by Euro20179** [reply]
Seems there is a fatal error in 2.1, you can either change line 620 from `_cur_pag_cur_page=$((i+1))` to `_cur_page=$((_cur_page+1))`, or you can upgrade to 2.2.alpha
```sh
git clone https://github.com/pystardust/ytfzf
cd ytfzf
sudo make install doc
```

**Reply by Quills245** [reply]
Ok, really thanks it worked!
